### PR TITLE
MEN-4422: Optionally install Mender Configure addon

### PR DESCRIPTION
**Pull request by lluiscampos**
Changelog: Support installing mender-configure addon. Not installed by
default, it can be configured using MENDER_ADDON_CONFIGURE_INSTALL and
MENDER_ADDON_CONFIGURE_VERSION variables.

Changelog: Set mender-configure version to master

### CODE REVIEW

**Review by kacf — APPROVED** [review]
The content looks good.

But how will this work together with the mender-image-tests test? Since it is not installed by default, I guess it will fail? How did we do it for mender-connect (if we did do it)?

File: mender-convert-modify [hunk 1]
```diff
@@ -141,6 +141,22 @@ if [ "${MENDER_ADDON_CONNECT_INSTALL}" = "y" ]; then
...
+
+  if [ "${MENDER_ADDON_CONFIGURE_VERSION}" = "latest" ]; then
+    deb_name=$(deb_from_repo_dist_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "stable" "mender-configure")
+  elif [ "${MENDER_ADDON_CONFIGURE_VERSION}" = "master" ]; then
+    deb_name=$(deb_from_repo_dist_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "experimental" "mender-configure")
+  else
+    DEBIAN_REVISION="-1"
+    deb_name=$(deb_from_repo_pool_get "work/deb-packages" ${MENDER_APT_REPO_URL} ${deb_arch} "mender-configure" "${MENDER_ADDON_CONFIGURE_VERSION}${DEBIAN_REVISION}")
+  fi
+
```
Comment (lluiscampos) [inline mender-convert-modify:156]: Note to self:

Missing creating `/data/mender-configure` dir and symlink `/var/lib/mender-configure`.

**Comment by lluiscampos** [comment]
> But how will this work together with the mender-image-tests test? Since it is not installed by default, I guess it will fail? How did we do it for mender-connect (if we did do it)?

Same as for `mender-connect`, we actually verify that the files *not* present. See https://github.com/mendersoftware/mender-image-tests/pull/39/files#diff-61f29111cea287297ab544f795cf48c3a8f83c2b5a24628fef969b8e65215772R203

**Comment by kacf** [comment]
Ah! I didn't spot that!

It's a shame we don't get to use the test to our advantage in mender-convert, but I agree it has to be like this, unless we want to make an explicit conversion just for tests (but this is a different beast altogether).

**Comment by lluiscampos** [comment]
> Ah! I didn't spot that!
> 
> It's a shame we don't get to use the test to our advantage in mender-convert, but I agree it has to be like this, unless we want to make an explicit conversion just for tests (but this is a different beast altogether).

This is somewhat related to tomorrow Backlog Grooming topic "Mender demo for `mender-convert`". I think that we should somewhere build (and test) an image with addons. We can briefly brainstorm on this tomorrow.

File: mender-convert-modify (hunk 1 above)
Comment (lluiscampos) [reply mender-convert-modify:156]: Done

**Comment by oleorhagen** [comment]
I'm sorry @lluiscampos but looking at this, I see that I've fucked up the formatting, last time. Please have a look at this, and then rebase if happy :pray:

**Comment by oleorhagen** [comment]
@lluiscampos You can now rebase, and merge :)

**Comment by lluiscampos** [comment]
> @lluiscampos You can now rebase, and merge :)

Done.

> 
> Sry for the hold-up. But github webhooks were acting up

No problem. Thank you for caring about formatting :)

**Comment by lluiscampos** [comment]
Also cherry-picked #306 to run tests together.

**Pull request merged**